pkg/proc/proc_test.go

```diff
@@ -2581,3 +2581,46 @@ func TestStacktraceWithBarriers(t *testing.T) {
 		}
 	})
 }
+
+func TestAttachDetach(t *testing.T) {
+	if runtime.GOOS == "darwin" {
+		// does not work on darwin
+		return
+	}
+	fixture := protest.BuildFixture("testnextnethttp")
+	cmd := exec.Command(fixture.Path)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	assertNoError(cmd.Start(), t, "starting fixture")
+
+	// wait for testnextnethttp to start listening
+	for {
+		conn, err := net.Dial("tcp", "localhost:9191")
+		if err == nil {
+			conn.Close()
+			break
+		}
+		time.Sleep(50 * time.Millisecond)
+	}
+
+	p, err := Attach(cmd.Process.Pid)
+	assertNoError(err, t, "Attach")
+	go func() {
+		time.Sleep(1 * time.Second)
+		http.Get("http://localhost:9191")
```

Comment:
This is just to make the p.Continue below hit a breakpoint, what happens afterwards doesn't really matter.
If the breakpoint isn't hit the test won't terminate because p.Continue doesn't return, FailNow called on a different goroutine doesn't work anyway.